### ISSUE TITLE
src/sysfiles.c warning at compilation

### ISSUE THREAD

**Issue by hungaborhorvath** [gap-system/gap#1303]
I am not sure if this is an error or not. When I compile most recent master with 
````
./autogen.sh
./configure
make
````
then I get the following warning message:
````
   C       src/sysfiles.c => obj/sysfiles.lo
src/sysfiles.c: In function ‘SyExecuteProcess’:
src/sysfiles.c:2990:30: warning: variable ‘func2’ might be clobbered by ‘longjmp’ or ‘vfork’ [-Wclobbered]
     sig_handler_t           *func2;
                              ^
````
This does not seem to harm the compilation process, and all other messages indicate that the compilation ended successfully. And indeed, gap can be started, no problems there. 

So I am unsure if this needs to be addressed or not. My system is debian jessie x64. I can provide the complete outputs if necessary.

**Comment by fingolfin** [comment]
Interesting. What C compiler is that (i.e. output of `cc -v`) ?

**Comment by hungaborhorvath** [comment]
````
ghorvath@SamsungLaptop:~$ cc -v
Using built-in specs.
COLLECT_GCC=cc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/4.9/lto-wrapper
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Debian 4.9.2-10' --with-bugurl=file:///usr/share/doc/gcc-4.9/README.Bugs --enable-languages=c,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-4.9 --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --with-gxx-include-dir=/usr/include/c++/4.9 --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --with-system-zlib --disable-browser-plugin --enable-java-awt=gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-4.9-amd64/jre --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-4.9-amd64 --with-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-4.9-amd64 --with-arch-directory=amd64 --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --enable-objc-gc --enable-multiarch --with-arch-32=i586 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 4.9.2 (Debian 4.9.2-10) 
````

**Comment by fingolfin** [comment]
I can't reproduce this with GCC 7.1 on Mac OS X. But perhaps this is Linux specific. Anyway, a fix for this should be to add a `volatile` to `func2`. If you do that, does the error go away? If yes, feel free to open a PR :-).

**Comment by hungaborhorvath** [comment]
Yes, the problem does go away, but somehow my Firefox is not willing to open the new branch I created to ask for a pull request. :-( I tried on another machine, no luck either. Is it some standard problem with github or is my Firefox acting up? Can I try a commandline version? The branch name is ````VolatileFunc2````.

**Comment by hungaborhorvath** [comment]
off: 
In fact, the Edit button or the Preview button does not seem to work either.....
on

**Comment by fingolfin** [comment]
It works fine for me in Firefox, which is my primary browser. What you write sounds a bit like network connection problems. Or perhaps you use NoScript and are blocking too much?

**Comment by hungaborhorvath** [comment]
off:
I only use adblock, but the problem remains even after I turned it off. And in any case, previously this was not a problem. 
on:
This issue is fixed in #1315.
